util/scheduled_executor.go

```diff
@@ -11,12 +11,13 @@ type ScheduledExecutor interface {
 }
 
 type scheduledExecutor struct {
-	period    time.Duration
-	cb        func()
-	isRunning AtomicBool
-	ticker    *time.Ticker
-	done      chan bool
-	wg        sync.WaitGroup
+	period             time.Duration
+	cb                 func()
+	isRunning          AtomicBool
+	ticker             *time.Ticker
```

Comment:
possible thought... what if we could pass in a configurable ticker. 

ex: 

```go
type Ticker interface {
	Stop()
	Reset()
	C()
}
```

We could then provide some common used tickers in this package (PlainTicker, PlainTickerInstantStart...), but it would allow the calling service to create their own (maybe I want a JitteringTicker, or a ticker NonLinearTicker)...

Reply:
i love this idea!